### PR TITLE
[vsphere] Ensure event.kind is correctly set for pipeline errors

### PR DESCRIPTION
**Pull request by MakoWish**
## What does this PR do?

Modify vsphere integration to correctly set `event.kind` for pipeline errors and ensure `error.message` is an array.

## Checklist

- [X] I have reviewed tips for building integrations and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's changelog.yml file.
- [x] I have verified that Kibana version constraints are current according to guidelines.
 
## Author's Checklist

- [x]  

## Related issues

* For #6582

## Screenshots

### CODE REVIEW

**Comment by MakoWish** [comment]
This one passes locally, and if I grep for "ignore_failure", the only instances are in fact Boolean. I am not sure why Jenkins is failing it for "ignore_failure" being string.

File: packages/vsphere/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -71,7 +71,10 @@ on_failure:
   - remove:
       field:
         - _tmp
-      ignore_failure: true
+      ignore_failure: tru
```
Comment (efd6) [inline packages/vsphere/data_stream/log/elasticsearch/ingest_pipeline/default.yml:74]:
```suggestion
      ignore_failure: true
```

Comment (MakoWish) [reply packages/vsphere/data_stream/log/elasticsearch/ingest_pipeline/default.yml:74]: <facepalm /> Thank you!

**Pull request closed without merging**